### DISCUSSION TITLE
how to verify if the code that i am writing is correct

### DISCUSSION THREAD

**Discussion by elec7rik** [Cyfrin/foundry-full-course-cu#733] [General]
is there any way to see patrick's notebooks to match with my code? it gets confusing sometimes to code while watching the video so i feel the need to verify my code. i couldn't find any notebooks in the repo. kindly let me know if it's there and what should i do if it's not. thanks.

**Comment by 0xTagger** [comment]
Not sure what you mean by notebooks, but all the final code associated with each lesson can be found in the repo. I agree it is difficult to pause and go back and forth sometimes, but that is part of the process when learning. For a lot of the lessons I would pause and code along side him and then rewatch the same video again so I can pay attention to the full lesson without pausing the second time. Hope this helps!

**Reply by elec7rik** [reply]
wow this may sound stupid but i can't find the associated codes in the repo. also, thanks for sharing the nice technique, i'll employ it.

**Reply by 0xTagger** [reply]
Of course! and no stupid questions here my friend! 

Use this link --> https://github.com/Cyfrin/foundry-full-course-f23 and scroll down to each lesson as you go. There will be a link for each lesson's respective code. 

Examples:

![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/114359465/c520e31e-db96-4c74-b6a1-c2c859871a90)
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/114359465/a844d147-a100-4f37-91d7-47e658cec398)

Let me know if there's anything else I can do to help :)

**Reply by elec7rik** [reply]
god i'm cracking up how can i not see this after scrolling multiple times through the list. thanks a bunch bud. this community is amazing.
